src/patternfly/components/Toolbar/toolbar.scss

```diff
@@ -221,30 +235,36 @@ $pf-v6--include-toolbar-breakpoints: true !default;
 
 // - Toolbar group
 .#{$toolbar}__group {
-  row-gap: var(--#{$toolbar}__group--RowGap, var(--#{$toolbar}--RowGap));
-  column-gap: var(--#{$toolbar}__group--ColumnGap, var(--#{$toolbar}--ColumnGap));
-  align-items: baseline;
-
-  // - Toolbar icon button group - Toolbar filter group
-  &.pf-m-icon-button-group {
-    column-gap: var(--#{$toolbar}__group--m-icon-button-group--ColumnGap, 0);
-  }
+  row-gap: var(--#{$toolbar}__group--RowGap);
+  column-gap: var(--#{$toolbar}__group--ColumnGap);
 
+  // - Toolbar filter group
   &.pf-m-filter-group {
     column-gap: var(--#{$toolbar}__group--m-filter-group--ColumnGap);
   }
 
-  // - Toolbar filter group - Toolbar label container
-  &.pf-m-filter-group,
-  &.pf-m-label-group-container {
-    flex-wrap: nowrap;
-  }
-
   // - Toolbar label group
   &.pf-m-label-group {
     flex: 1;
+    column-gap: var(--#{$toolbar}__group--m-label-group--ColumnGap);
+  }
+
+  // - Toolbar action group
+  &.pf-m-action-group {
+    column-gap: var(--#{$toolbar}__group--m-action-group--ColumnGap);
```

Comment:
the global gap spacers on the action group needs to be updated to: global/spacer/gap/action-to-action: 16px/spacer-300 
👍